server/client.go

```diff
@@ -15,7 +16,14 @@ import (
 
 // Client is a new HTTP Client to talk to the Skype server
 type Client struct {
-	httpClient *http.Client
+	httpClient  *http.Client
+	logger      Logger
+	logRequests bool
+}
+
+// Logger is used for logging requests
+type Logger interface {
+	LogInfo(msg string, keyValuePairs ...interface{})
```

Comment:
I suppose `INFO` makes sense, because if they want to shut this off, they can just turn off the plugin's setting. Versus having to enable/disable `DEBUG` logging and potentially have a bit more spam in the mix in the logs.

Can we have this `Logger` have all methods (`LogError`, `LogWarn` `LogDebug`), then use the logger in any place we are logging things in the plugin?

Reply:
I'm a little confused. Why use this `Logger` for all logging?

Reply:
We've developed a pattern for plugins to have a separation of concerns between logging etc. Especially since we are introducing the `Logger` concept to this plugin in this PR, I think it makes sense to "replace" the direct usage of `p.API.Log*` with calls to this `Logger`, as we've done in other plugins.